tests/test_sknnr.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import xarray as xr
+from sknnr import GNNRegressor
+
+from sknnr_spatial import wrap
+
+from .image_utils import parametrize_model_data
+
+
+@parametrize_model_data(image_types=(xr.DataArray,))
+def test_kneighbors_returns_df_index(model_data):
+    """Test that sknnr estimators return dataframe indices."""
+    # Create dummy plot data
+    X = np.random.rand(10, 3) + 10.0
+    y = np.random.rand(10, 3)
+
+    # Create an image of zeros and set the first plot to zeros to ensure that the
+    # first index is the nearest neighbor to all pixels
+    X_image = np.zeros((2, 2, 3))
+    X[0] = [0, 0, 0]
+
+    # Convert model data to the correct types
+    X_image, X, y = model_data.set(
+        X_image=X_image,
+        X=X,
+        y=y,
+    )
+
+    # Offset the dataframe index to differentiate it from the array index
+    df_index_offset = 999
+    X.index += df_index_offset
+
+    est = wrap(GNNRegressor()).fit(X, y)
+    idx = est.kneighbors(X_image, return_distance=False, return_dataframe_index=False)
+    df_idx = est.kneighbors(X_image, return_distance=False, return_dataframe_index=True)
+
+    assert idx.shape == df_idx.shape
+
+    assert idx.min().compute() == 0
+    assert df_idx.min().compute() == df_index_offset
```

Comment:
I *think* I'm reading your test correctly, but check me if I'm wrong or if this is unimportant to the test.  Wouldn't you expect not only the `min`, but all pixels to equate to either 0 or `df_index_offset`?  Is that worth testing for?

Reply:
Yes, more thorough and probably more understandable to check all the values in the first neighbor!

Reply:
Looks great!